analyzers/tests/SonarAnalyzer.UnitTest/Rules/ClassAndMethodNameTest.cs

```diff
@@ -94,27 +94,19 @@ public class ClassAndMethodNameTest
                 .AddReferences(TestHelper.ProjectTypeReference(projectType))
                 .WithOptions(ParseOptionsHelper.FromCSharp8).Verify();
 
-        [TestMethod]
-        public void TestSplitToParts() =>
-            new[]
-            {
-                ("foo", new[] { "foo" }),
-                ("Foo", new[] { "Foo" }),
-                ("FFF", new[] { "FFF" }),
-                ("FfF", new[] { "Ff", "F" }),
-                ("Ff9F", new[] { "Ff", "9", "F" }),
-                ("你好", new[] { "你", "好" }),
-                ("FFf", new[] { "F", "Ff" }),
-                (string.Empty,  Array.Empty<string>()),
-                ("FF9d", new[] { "FF", "9", "d" }),
-                ("y2x5__w7", new[] { "y", "2", "x", "5", "_", "_", "w", "7" }),
-                ("3%c#account", new[] { "3", "%", "c", "#", "account" }),
-            }
-            .Select(x =>
-            (
-                actual: CS.ClassAndMethodName.SplitToParts(x.Item1).ToArray(),
-                expected: x.Item2))
-            .ToList()
-            .ForEach(x => x.actual.Should().Equal(x.expected));
+        [DataTestMethod]
```

Comment:
Just a refactoring. The test uses the right patterns for such a kind of test now.